src/test/java/org/gridsuite/study/server/NetworkModificationTest.java

```diff
@@ -2444,79 +2433,108 @@ public void tearDown() throws IOException {
+            requestBody);
     }
 
-    private void verifyPostWithBody(String requestBody) {
-        verifyPostWithBody(1, requestBody);
+    private void verifyNetworkModificationPut(UUID stubId, String requestBody) {
+        verifyPutRequest(stubId, "/v1/network-modifications/" + MODIFICATION_UUID, false, Map.of(), requestBody);
     }
 
-    private void verifyPostWithBody(Integer count, String requestBody) {
-        wireMock.verify(count, WireMock.postRequestedFor(WireMock.urlPathEqualTo(
-                        "/v1/network-modifications"))
-                .withQueryParam("networkUuid", WireMock.equalTo(NETWORK_UUID_STRING))
-                .withQueryParam("groupUuid", WireMock.matching(".*"))
-                .withRequestBody(WireMock.equalToJson(requestBody)));
+    private void verifyPostRequest(UUID stubId, String url, Map<String, StringValuePattern> queryParams) {
+        verifyPostRequest(stubId, url, false, queryParams, null);
     }
 
-    private void verifyPostWithBodyAndVariant(String requestBody, String variantId) {
-        verifyPostWithBodyAndVariant(1, requestBody, variantId);
+    private void verifyPostRequest(UUID stubId, String url, boolean urlMatching, Map<String, StringValuePattern> queryParams, String body) {
+        RequestPatternBuilder requestBuilder = urlMatching ? WireMock.postRequestedFor(WireMock.urlMatching(url)) : WireMock.postRequestedFor(WireMock.urlPathEqualTo(url));
+        verifyRequest(stubId, requestBuilder, queryParams, body);
     }
 
-    private void verifyPostWithBodyAndVariant(Integer count, String requestBody, String variantId) {
-        wireMock.verify(count, WireMock.postRequestedFor(WireMock.urlPathEqualTo(
-                        "/v1/network-modifications"))
-                .withQueryParam("networkUuid", WireMock.equalTo(NETWORK_UUID_STRING))
-                .withQueryParam("groupUuid", WireMock.matching(".*"))
-                .withQueryParam("variantId", WireMock.equalTo(variantId))
-                .withRequestBody(WireMock.equalToJson(requestBody)));
+    private void verifyPutRequestWithUrlMatching(UUID stubId, String url, Map<String, StringValuePattern> queryParams, String body) {
+        verifyPutRequest(stubId, url, true, queryParams, body);
     }
 
-    private void verifyPutWithBody(String requestBody) {
-        wireMock.verify(1, WireMock.putRequestedFor(WireMock.urlPathEqualTo(
-                        "/v1/network-modifications/" + MODIFICATION_UUID))
-                .withRequestBody(WireMock.equalToJson(requestBody)));
+    private void verifyPutRequest(UUID stubId, String url, boolean urlMatching, Map<String, StringValuePattern> queryParams, String body) {
+        RequestPatternBuilder requestBuilder = urlMatching ? WireMock.putRequestedFor(WireMock.urlMatching(url)) : WireMock.putRequestedFor(WireMock.urlPathEqualTo(url));
```

Comment:
```suggestion
        RequestPatternBuilder requestBuilder = urlMatching ? WireMock.putRequestedFor(WireMock.urlPathMatching(url)) : WireMock.putRequestedFor(WireMock.urlPathEqualTo(url));
```

Reply:
For the 2 others too

Reply:
?

Reply:
Done

Reply:
it was already like that !

Reply:
I guess some params are missing, we can't fix it ?

Reply:
I'm checking

Reply:
Fixed